homeassistant/components/system_bridge/media_source.py

```diff
@@ -0,0 +1,172 @@
+"""System Bridge Media Source Implementation."""
+from __future__ import annotations
+
+import asyncio
+
+import async_timeout
+from systembridgeconnector.models.media_directories import MediaDirectories
+from systembridgeconnector.models.media_files import File as MediaFile
+
+from homeassistant.components.media_player.const import MEDIA_CLASS_DIRECTORY
+from homeassistant.components.media_source.const import (
+    MEDIA_CLASS_MAP,
+    MEDIA_MIME_TYPES,
+)
+from homeassistant.components.media_source.models import (
+    BrowseMediaSource,
+    MediaSource,
+    MediaSourceItem,
+    PlayMedia,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_API_KEY, CONF_HOST, CONF_PORT
+from homeassistant.core import HomeAssistant
+
+from .const import DOMAIN
+from .coordinator import SystemBridgeDataUpdateCoordinator
+
+
+async def async_get_media_source(hass: HomeAssistant) -> MediaSource:
+    """Set up SystemBridge media source."""
+    for entry in hass.config_entries.async_entries(DOMAIN):
```

Comment:
Should replace this with a "System Bridge" title and list each available entry as children.

Currently, this will only show a media source for the first available/loaded entry/device, where we want all available systems to be listed

Reply:
Done